src/components/tooltip/tooltip.js

```diff
@@ -10,6 +37,123 @@ import {h, Component} from 'preact';
  * @extends {Component}
  */
 class Tooltip extends Component {
+  state: Object;
+  _hoverTimeout: ?number;
+  textElement: HTMLSpanElement;
+  lastAlternativeTypeIndex: number = -1;
+
+  /**
+   * default component props
+   * @type {Object}
+   * @memberof Tooltip
+   */
+  static defaultProps = {
+    type: ToolTipType.Top,
+    maxWidth: '240px'
+  };
+
+  /**
+   * @constructor
+   * @param {*} props props
+   * @memberof Tooltip
+   */
+  constructor(props: Object) {
+    super(props);
+  }
+
+  /**
+   * on mouse over handler.
+   * @memberof Tooltip
+   * @returns {void}
+   */
+  onMouseOver(): void {
+    if (this._hoverTimeout) {
+      clearTimeout(this._hoverTimeout);
+      this._hoverTimeout = null;
+    }
+    this._hoverTimeout = setTimeout(() => {
+      this.setState({showTooltip: true});
+    }, TOOLTIP_SHOW_TIMEOUT);
+  }
+
+  /**
+   * on mouse leave handler.
+   * @memberof Tooltip
+   * @returns {void}
+   */
+  onMouseLeave(): void {
+    this.setState({showTooltip: false});
+    clearTimeout(this._hoverTimeout);
+    this._hoverTimeout = null;
+  }
+
+  /**
+   * brings another tooltip type which hasn't been marked as invalid
+   * @memberof Tooltip
+   * @returns {string} tooltip type
+   */
+  getAlternateType(): ?string {
+    return ((Object.values(ToolTipType).find((item, index) => {
```

Comment:
`const ToolTipTypeValues = Object.values(ToolTipType)
ToolTipTypeValues.length > this.lastAlternativeTypeIndex ? this.lastAlternativeTypeIndex++ : this.lastAlternativeTypeIndex
return ToolTipTypeValues[this.lastAlternativeTypeIndex]`

Reply:
i am leaving it as it is